install.sh

```diff
@@ -0,0 +1,877 @@
+
+    # --- use service or environment location depending on systemd/openrc ---
+    if [ "${HAS_SYSTEMD}" = true ]; then
+        FILE_RKE2_SERVICE=${SYSTEMD_DIR}/${SERVICE_RKE2}
+        FILE_RKE2_ENV=${SYSTEMD_DIR}/${SERVICE_RKE2}.env
+    elif [ "${HAS_OPENRC}" = true ]; then
+        ${SUDO} mkdir -p /etc/rancher/rke2
+        FILE_RKE2_SERVICE=/etc/init.d/${SYSTEM_NAME}
+        FILE_RKE2_ENV=/etc/rancher/rke2/${SYSTEM_NAME}.env
+    fi
+
+    # --- get hash of config & exec for currently installed rke2 ---
+    PRE_INSTALL_HASHES=$(get_installed_hashes)
+
+    # --- if bin directory is read only skip download ---
+    if [ "${INSTALL_RKE2_BIN_DIR_READ_ONLY}" = true ]; then
+        INSTALL_RKE2_SKIP_DOWNLOAD=true
+    fi
+
+    # --- setup channel values
+    INSTALL_RKE2_CHANNEL_URL=${INSTALL_RKE2_CHANNEL_URL:-"https://update.rke2.io/v1-release/channels"}
+    INSTALL_RKE2_CHANNEL=${INSTALL_RKE2_CHANNEL:-"stable"}
+}
+
+# can_skip_download checks if skip download 
+# environment variable set.
+can_skip_download() {
+    if [ "${INSTALL_RKE2_SKIP_DOWNLOAD}" != true ]; then
+        return 1
+    fi
+}
+
+# verify_rke2_is_executable verify an executabe 
+# rke2 binary is installed.
+verify_rke2_is_executable() {
+    if [ ! -x ${BIN_DIR}/rke2 ]; then
+        fatal "executable rke2 binary not found at ${BIN_DIR}/rke2"
+    fi
+}
+
+# setup_verify_arch set arch and suffix, 
+# fatal if architecture not supported.
+setup_verify_arch() {
+    if [ -z "${ARCH}" ]; then
+        ARCH=$(uname -m)
+    fi
+    case ${ARCH} in
+        amd64)
+            ARCH=amd64
+            SUFFIX=
+            ;;
+        x86_64)
+            ARCH=amd64
+            SUFFIX=
+            ;;
+        arm64)
+            ARCH=arm64
+            SUFFIX=-${ARCH}
+            ;;
+        aarch64)
+            ARCH=arm64
+            SUFFIX=-${ARCH}
+            ;;
+        arm*)
+            ARCH=arm
+            SUFFIX=-${ARCH}hf
+            ;;
+        *)
+            fatal "unsupported architecture ${ARCH}"
+    esac
+}
+
+# verify_downloader verifies existence of
+# network downloader executable.
+verify_downloader() {
+    cmd="$(which $1)"
+    if [ -z ${cmd} ]; then
+        return 1
+    fi
+    if [ ! -x ${cmd} ]; then
+        return 1
+    fi
+
+    # Set verified executable as our downloader program and return success
+    DOWNLOADER=${cmd}
+    return 0
+}
+
+# setup_tmp creates a tempory directory 
+# and cleans up when done.
+setup_tmp() {
+    TMP_DIR=$(mktemp -d -t rke2-install.XXXXXXXXXX)
+    TMP_HASH=${TMP_DIR}/rke2.hash
+    TMP_BIN=${TMP_DIR}/rke2.bin
+    cleanup() {
+        code=$?
+        set +e
+        trap - EXIT
+        rm -rf "${TMP_DIR}"
+        exit $code
+    }
+    trap cleanup INT EXIT
+}
+
+# --- use desired rke2 version if defined or find version from channel ---
+get_release_version() {
+    if [ -n "${INSTALL_RKE2_COMMIT}" ]; then
+        VERSION_RKE2="commit ${INSTALL_RKE2_COMMIT}"
```

Comment:
I guess we should either set up the same commit storage space for rke2 like k3s or skip this option

Reply:
I agree. :D